executor/hash_table.go

```diff
@@ -182,14 +182,18 @@ func (c *hashRowContainer) GetAllMatchedRows(probeHCtx *hashContext, probeSideRo
 		//     (  ? , 1, 2), that exactly with value as 1 and 2 in the second and third join key column.
 		needCheckProbeRowPos = needCheckProbeRowPos[:0]
 		needCheckBuildRowPos = needCheckBuildRowPos[:0]
+		needCheckBuildTypes = needCheckBuildTypes[:0]
+		needCheckProbeTypes = needCheckProbeTypes[:0]
 		keyColLen := len(c.hCtx.naKeyColIdx)
 		for i := 0; i < keyColLen; i++ {
 			// since all bucket is from hash table (Not Null), so the buildSideNullBits check is eliminated.
 			if probeKeyNullBits.UnsafeIsSet(i) {
 				continue
 			}
 			needCheckBuildRowPos = append(needCheckBuildRowPos, c.hCtx.naKeyColIdx[i])
+			needCheckBuildTypes = append(needCheckBuildTypes, c.hCtx.allTypes[i])
 			needCheckProbeRowPos = append(needCheckProbeRowPos, probeHCtx.naKeyColIdx[i])
+			needCheckProbeTypes = append(needCheckProbeTypes, probeHCtx.allTypes[i])
```

Comment:
The name `needCheckBuildRowPos` and `needCheckProbeRowPos` confuses me. They save the column index, not the row position.

Reply:
renamed